cucumber-messages/messages.proto

```diff
@@ -484,6 +490,7 @@ message TestResult {
   string message = 2;
   reserved 3;
   Duration duration = 4;
+  bool retried = 5;
```

Comment:
I would expect this to be `attemptNumber`.

Reply:
The TestCaseFinished will already have the attemptNumber. This is the result which for TestCaseFinished now has the additional property retried to signal if the testCase will have another attempt

Reply:
Then I don't think this field is required then. It is implied by `attemptNumber > 1` in `TestCaseFinished` which you need to go through to get to the `TestResult`.

Reply:
Consider this situation, attempt 1 fails. We have no way to know if that test case will be retried or not. That information is important in order to know if this failure should cause the suite to fail. If its going to be retried, the result can essentially be ignored (cucumber-js currently outputs it as a warning). When attempt numbers are greater than 1, sure that gives an indication that earlier attempts are retried, but I want that information as part of attempt 1 in its test case finished event

Reply:
So this flag means the scenario will be retried then?

Reply:
I dont know if that info should be part of the stream. A test run isnt done untill the finished event.  And a listener shouldnt have to know the rules to decide when a run was passed. All that info should already be there.

So ideally we'd have a retried status to indicate the test failed and will be retried.

Reply:
Ideally reporters just list the outcomes in a nice way.

Reply:
Correct, the flag means the test case will be retried. I've already been through a couple long debate on a few PRs about whether or not cucumber-js was going to introduce a RETRIED status (something I was arguing for). Link to one discussion which also mentions the older one: https://github.com/cucumber/cucumber-js/pull/1205#discussion_r290336731. We ended with the idea of keeping the failed status and having a retried flag.

I agree the listener shouldn't need to know the rules about if it was passed, failed, retried and that is what this flag is for. 

Thinking on this more, probably worth changing the name of this to `will_be_retried`

Reply:
That would be a better name!

Another thing I am concerned about here is that the format of the message requires a cucumber implementation to immediatly decide the retry status.

This pushes the retry mechanism into the execution of a scenario. It could be a much higher level decision.

Could you point me to the code in CucumberJS that makes this decision?

Reply:
Updated. 

So in cucumber-js, the highest level runner determines the max attempts for each test case and then the test case runner takes that in and will retry if the scenario failed and the current attempt number is less than the max attempts.

Determining max attempts for each test case:
https://github.com/cucumber/cucumber-js/blob/master/src/runtime/helpers.js#L43
Determining if a test case should retry
https://github.com/cucumber/cucumber-js/blob/master/src/runtime/test_case_runner.js#L190